src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/resource/LocationOccupancyResource.kt

```diff
@@ -0,0 +1,121 @@
+import uk.gov.justice.digital.hmpps.locationsinsideprison.service.Prisoner
+import java.util.*
+
+@RestController
+@Validated
+@RequestMapping("/location-occupancy", produces = [MediaType.APPLICATION_JSON_VALUE])
+@Tag(
+  name = "Location Occupancy",
+  description = "Returns location information with occupancy information",
+)
+class LocationOccupancyResource(
+  private val locationService: LocationService,
+) {
+
+  @PreAuthorize("hasRole('ROLE_VIEW_LOCATIONS')")
+  @GetMapping("/cells-with-capacity/{prisonId}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "List of cells by group at prison which have capacity.",
+    description = "Requires role VIEW_LOCATIONS",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns cells with capacity available",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the VIEW_LOCATIONS role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getCellsWithCapacity(
+    @Schema(description = "Prison Id", example = "MDI", required = true, minLength = 3, maxLength = 5, pattern = "^[A-Z]{2}I|ZZGHI$")
+    @Size(min = 3, message = "Prison ID cannot be blank")
+    @Size(max = 5, message = "Prison ID must be 3 characters or ZZGHI")
+    @Pattern(regexp = "^[A-Z]{2}I|ZZGHI$", message = "Prison ID must be 3 characters or ZZGHI")
+    @PathVariable prisonId: String,
+    @Schema(description = "Location Id in the prison below which to find cells", example = "de91dfa7-821f-4552-a427-bf2f32eafeb0", required = false)
+    @RequestParam(name = "locationId", required = false) locationId: UUID? = null,
+    @Schema(description = "Group name for a sub location to find cells", example = "Wing A", required = false)
+    @RequestParam(name = "groupName", required = false) groupName: String? = null,
+    @Schema(description = "Only return cells of a specified specialist type", example = "CSU", required = false)
+    @RequestParam(name = "specialistCellType", required = false) specialistCellType: SpecialistCellType? = null,
+    @Schema(description = "Include prisoner details in this cell", required = false, defaultValue = "false")
+    @RequestParam(name = "includePrisonerInformation", required = false, defaultValue = "false") includePrisonerInformation: Boolean = false,
+  ): List<CellWithSpecialistCellTypes> =
```

Comment:
is it not required to publish an event for this?

Reply:
No because it doesn't change any data.  Only "writes" cause an event